File: src/main/java/javafiles/dataaccessfilesTEMP/JSONIOBuilder.java
package javafiles.dataaccessfilesTEMP;

import javafiles.customexceptions.ReadWriteException;

class JSONIOBuilder implements FileIOReaderBuilder, FileIOWriterBuilder {
    private final String[] EXTENSIONS;

    JSONIOBuilder(String[] extensions) {
        EXTENSIONS = extensions;
    }

    @Override
    public FileIO createFileIOReader(String path) throws ReadWriteException {
        return new JSONIO(path, 'r');
    }

    @Override
    public FileIO createFileIOWriter(String path) throws ReadWriteException {
        return new JSONIO(path, 'w');
    }

    @Override
    public String[] getExtensions() {
        return EXTENSIONS;
    }
}

File: src/main/java/javafiles/dataaccessfiles/builderimplements/JSONIOBuilder.kt
package javafiles.dataaccessfiles.builderimplements

import javafiles.customexceptions.ReadWriteException
import javafiles.dataaccessfiles.*
import javafiles.dataaccessfiles.fileioimplements.JSONIORead
import javafiles.dataaccessfiles.fileioimplements.JSONIOWrite
import java.io.File

internal class JSONIOBuilder(override val extensions: Array<String>) : FileIOReaderBuilder, FileIOWriterBuilder {
    @Throws(ReadWriteException::class)
    override fun createFileIOReader(path: String): FileIO {
        return JSONIORead(File(path))
    }

    @Throws(ReadWriteException::class)
    override fun createFileIOWriter(path: String): FileIO {
        return JSONIOWrite(File(path))
    }
}
